windows-packaging/CalicoWindows/libs/calico/calico.psm1

```diff
@@ -413,6 +413,47 @@ function Set-MetaDataServerRoute($mgmtIP)
     }
 }
 
+function Get-ContainerdCniBinDir()
+{
+    $config = getContainerdConfig
+    return $config['plugins."io.containerd.grpc.v1.cri".cni']['bin_dir']
+}
+
+function Get-ContainerdCniConfDir()
+{
+    $config = getContainerdConfig
+    return $config['plugins."io.containerd.grpc.v1.cri".cni']['conf_dir']
+}
+
+function getContainerdService()
+{
+    # Don't use get-wmiobject since that is not available in Powershell 7.
+    return Get-CimInstance -Query "SELECT * from Win32_Service WHERE name = 'containerd'"
+}
+
+function getContainerdConfig()
+{
+    if (!(Get-InstalledModule PsIni))
+    {
+        Write-Host "Dependency PsIni not installed, installing..."
+        Install-module PsIni -Confirm:$False -Force

```

Comment:
I'm not sure what is the security implication of importing a third party module. Can we do it without this module?

Reply:
We discussed this and agree to remove this dependency and instead use a sensible default.